cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -169,7 +170,7 @@ public ApplicationId start(NamespaceId namespace, AppRequest<?> appRequest) thro
     Set<String> realDatasets = appRequest.getPreview() == null ? new HashSet<String>()
       : appRequest.getPreview().getRealDatasets();
 
-    ApplicationId previewApp = namespace.app(RunIds.generate().getId());
+    ApplicationId previewApp = namespace.app(PREFIX + System.currentTimeMillis());
```

Comment:
I think it is good to give preview a more meaningful name. An UUID will make the user confused. They may consider this as a runid.